### DISCUSSION TITLE
`Click for preview` for blocked resources?

### DISCUSSION THREAD

**Discussion by MasterKia** [uBlockOrigin/uBlock-issues#2578] [Ideas] [answered]
Some sites have a lot of advertisement `png` and `gif` images on the same path, so I usually do something like:
`||dl.konkur.in/image/ads^$image,1p`

It happens that sometimes there are a couple of non-ads `png` images hosted on that path too. Then the logger could be used to view those images and add exceptions for those images, if needed.

A normal resource having "Click to preview":
![Screenshot_20230402_225540](https://user-images.githubusercontent.com/17685483/229374528-06035d38-79ad-48f4-868f-a241d8e5f65b.png)

A blocked resource not having "Click to preview":
![Screenshot_20230402_225731](https://user-images.githubusercontent.com/17685483/229374610-5aa95ae6-524e-4419-8f1a-c40dbe3c93fb.png)

I don't know if that's trivial to implement.

**Comment by gorhill** [comment, marked as answer]
Just click the link in the logger, or select the URL, right-click to open in a new tab.

**Reply by MasterKia** [reply]
Ok, thank you. I've been doing that, it was just `Click for preview` is quicker and more convenient.

**Reply by gorhill** [reply]
https://github.com/gorhill/uBlock/issues/1999

**Reply by MasterKia** [reply]
From what I understand, back then the logger entry _automatically_ loaded any images, blocked or not. Leading to the user complaining and thinking that the image was not in fact blocked even though the logger entry shows the image as blocked. 

So you added this:
> when the preview area is visible, the user must "click to preview".

Now the logger entry won't automatically load the image and the user won't think that the blocked image is not blocked, which in fact is blocked.
Acknowledging that; now the user can make the choice to press "Click for preview" to load the image by itself.

> I removed the ability to preview for network requests which were blocked

Would bringing that back in the current state cause confusion like in https://github.com/gorhill/uBlock/issues/1999?

**Reply by gorhill** [reply]
In the thread I also say that the solution is to fetch from the page world, this is _not_ trivial and brings it's own complications (the image might actually be a resource of an embedded frame). Just open the image as a document, it's simpler for me, and there is also the fact I just don't want to spend time on this.
